src/block/BaseSign.php

```diff
@@ -94,6 +102,35 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 		return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
 	}
 
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if($player !== null and $item instanceof Dye){
+			$dyeColor = $item->getColor();
+
+			$oldColor = $this->textColor;
+
+			if($dyeColor->equals($oldColor)){
+				return false;
+			}
+
+			$this->textColor = $dyeColor;
+
+			$this->position->getWorld()->setBlock($this->position, $this);
+
+			$item->pop();
+			return true;
```

Comment:
This block could use less whitespace. It's quite spaced out and difficult to read.